core-services/default-data-handler/src/main/java/org/egov/handler/service/DataHandlerService.java

```diff
@@ -0,0 +1,317 @@
+
+        // Search data for the schema code in default tenetId
+        List<Mdms> mdmsList = getAllMdmsResults(serviceConfig.getDefaultTenantId(), TENANT_BOUNDARY_SCHEMA, requestInfo);
+        // Create schema data in the given tenantId
+        for (Mdms mdms : mdmsList) {
+            mdms.setTenantId(targetTenantId);
+
+            JsonNode dataNode = mdms.getData();
+            if (dataNode.has("boundary")) {
+                // Cast the 'boundary' node to ObjectNode so that we can modify it
+                ObjectNode boundaryNode = (ObjectNode) dataNode.get("boundary");
+
+                // Modify the 'code' field within the 'boundary' node
+                boundaryNode.put("code", targetTenantId);
+
+                // Set the modified 'data' back to the Mdms object (optional, since it's mutable)
+                ((ObjectNode) dataNode).set("boundary", boundaryNode);
+                mdms.setData(dataNode);
+            } else {
+                log.info("Boundary node does not exist in the data.");
+            }
+            MdmsRequest mdmsRequest = MdmsRequest.builder().requestInfo(requestInfo).mdms(mdms).build();
+            log.info("{} : {}", mdms.getSchemaCode(), mdms.getUniqueIdentifier());
+            mdmsV2Util.createMdmsData(mdmsRequest);
+        }
+    }
+
+    // Method to get all search results with pagination
+    public List<Mdms> getAllMdmsResults(String tenantId, String schemaCode, RequestInfo requestInfo) {
+        List<Mdms> allMdmsResults = new ArrayList<>();
+        int limit = 100; // Default limit
+        int offset = 0; // Default offset
+
+        while (true) {
+            // Create MdmsCriteriaV2 with current offset and limit
+            MdmsCriteriaV2 mdmsCriteria = MdmsCriteriaV2.builder().tenantId(tenantId).schemaCode(schemaCode).offset(offset).limit(limit).build();
+
+            MdmsCriteriaReqV2 mdmsCriteriaReq = MdmsCriteriaReqV2.builder().requestInfo(requestInfo).mdmsCriteria(mdmsCriteria).build();
+
+            // Fetch results from the repository
+            MdmsResponseV2 dataSearchResponse = mdmsV2Util.searchMdmsData(mdmsCriteriaReq);
+            List<Mdms> mdmsList = dataSearchResponse.getMdms();
+
+            // Add the current batch of results to the list
+            allMdmsResults.addAll(mdmsList);
+
+            // Check if there are fewer results than the limit; if so, this is the last page
+            if (mdmsList.size() < limit) {
+                break;
+            }
+
+            // Update offset for the next batch
+            offset += limit;
+        }
+        return allMdmsResults;
+    }
+
+    public void createTenantConfig(TenantRequest tenantRequest) {
+        TenantConfigResponse tenantConfigSearchResponse = tenantManagementUtil.searchTenantConfig(serviceConfig.getDefaultTenantId(), tenantRequest.getRequestInfo());
+        List<TenantConfig> tenantConfigList = tenantConfigSearchResponse.getTenantConfigs();
```

Comment:
You should check if response is not empty and lenght is equal to 1 else throw error. Do get(0) and directly call tenantConfig create instead od running for loop